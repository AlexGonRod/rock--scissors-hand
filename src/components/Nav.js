import React from 'react';
import { Link } from "react-router-dom";

const Nav = ({ user }) => {
	return (
		<div className="nav">
			<div className="user-text">
                <span>{user}</span>
			</div>
            <Link to="/" className="back-nav"/>
		</div>
	)
}

export default Nav
