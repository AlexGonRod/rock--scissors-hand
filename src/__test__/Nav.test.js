import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import Nav from '../components/Nav';

describe('<Nav />', () => {
    let wrapper = shallow(<Nav />);
    beforeEach(() => {
        wrapper = shallow(<Nav />);
    });
    test('<Nav /> renders', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('<Nav /> user prop', () => {
        const user='test-user';
        
        const wrapper = shallow(<Nav user={user}/>)
        expect(wrapper.find('.user-text span').length).toEqual(1);
    })
});