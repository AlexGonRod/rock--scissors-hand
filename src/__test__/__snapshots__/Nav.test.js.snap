// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> <Nav /> renders 1`] = `
<div
  className="nav"
>
  <div
    className="user-text"
  >
    <span />
  </div>
  <Link
    className="back-nav"
    to="/"
  />
</div>
`;
